### PR TITLE
Add URL parameters to ES output

### LINKED ISSUE #805: Elasticsearch output HTTP parameters
Add support for additional HTTP parameters in elasticsearch output for use with node ingest.

### PR DESCRIPTION
**Pull request by urso**
Add config file option to elasticsearch output to pass additional parameters
with index operations.

Closes #805

### CODE REVIEW

**Comment by monicasarbu** [comment]
LGTM

File: filebeat/filebeat.yml [hunk 1]
```diff
@@ -151,6 +151,9 @@ filebeat:
   # Event count spool threshold - forces network flush if exceeded
   #spool_size: 2048
 
+  # Enable async publisher pipeline in filebeat (Experimental!)
+  #publish_async: false
```
Comment (ruflin) [inline filebeat/filebeat.yml]: This will probably disappear after rebase: https://github.com/elastic/beats/blob/master/filebeat/filebeat.yml#L155

**Comment by ruflin** [comment]
LGTM. Needs rebase.

File: filebeat/filebeat.yml (hunk 1 above)
Comment (urso) [inline filebeat/filebeat.yml]: this will disappear on purpose?

Comment (ruflin) [inline filebeat/filebeat.yml]: By disappear I mean: not part of the diff as it is already in master :-)

**Comment by Bargs** [comment]
Could we document this config option in the Filebeat docs as well? I'll need docs to point users to from the Add Data UI so they can set up filebeat to send data through their ingest pipeline. As far as I can tell, `libbeat/docs/outputconfig.asciidoc` doesn't show up in the docs on the Elastic website anywhere.

**Comment by dedemorton** [comment]
@Bargs outputconfig.asciidoc is a shared file. We pull the content of that file into the the documentation for all of the Beats (see the topic called Configuration Options). For example, you'll see the content here for Filebeat: https://www.elastic.co/guide/en/beats/filebeat/1.1/filebeat-configuration-details.html#configuration-output. 

Of course, the link points to version 1.1, so you won't see the the parameters option, but you get the point.

**Pull request merged**